src/main/java/com/refinedmods/refinedstorage/network/OpenNetworkItemMessage.java

```diff
@@ -4,37 +4,62 @@
 import com.refinedmods.refinedstorage.apiimpl.network.grid.factory.PortableGridGridFactory;
 import com.refinedmods.refinedstorage.item.NetworkItem;
 import com.refinedmods.refinedstorage.item.blockitem.PortableGridBlockItem;
+import net.minecraft.client.Minecraft;
 import net.minecraft.entity.player.ServerPlayerEntity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.common.util.LazyOptional;
 import net.minecraftforge.fml.network.NetworkEvent;
+import top.theillusivec4.curios.api.CuriosApi;
+import top.theillusivec4.curios.api.type.capability.ICuriosItemHandler;
+import top.theillusivec4.curios.api.type.inventory.ICurioStacksHandler;
 
+import java.util.Optional;
 import java.util.function.Supplier;
 
 public class OpenNetworkItemMessage {
     private final int slotId;
+    private final String curioSlot;
 
-    public OpenNetworkItemMessage(int slotId) {
+    public OpenNetworkItemMessage(int slotId, String curioSlot) {
         this.slotId = slotId;
+        this.curioSlot = curioSlot;
     }
 
     public static OpenNetworkItemMessage decode(PacketBuffer buf) {
-        return new OpenNetworkItemMessage(buf.readInt());
+        return new OpenNetworkItemMessage(buf.readInt(), buf.readString(32767));
     }
 
     public static void encode(OpenNetworkItemMessage message, PacketBuffer buf) {
         buf.writeInt(message.slotId);
+        buf.writeString(message.curioSlot);
     }
 
     public static void handle(OpenNetworkItemMessage message, Supplier<NetworkEvent.Context> ctx) {
         ServerPlayerEntity player = ctx.get().getSender();
 
         if (player != null) {
             ctx.get().enqueueWork(() -> {
-                ItemStack stack = player.inventory.getStackInSlot(message.slotId);
+                ItemStack stack = null;
+                if (message.curioSlot.isEmpty()) {
+                    stack = player.inventory.getStackInSlot(message.slotId);
+                } else {
+                    LazyOptional<ICuriosItemHandler> curiosHandler = CuriosApi.getCuriosHelper().getCuriosHandler(player);
```

Comment:
Please split this code a bit more up, it's too much lines / too much indentation in a single method.